cockatrice/src/tab_deck_editor.cpp

```diff
@@ -863,6 +870,55 @@ void TabDeckEditor::actOpenCustomsetsFolder() {
 
 }
 
+void TabDeckEditor::actAddCustomSet()
+{
+
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
+    QString dataDir = QDesktopServices::storageLocation(QDesktopServices::DataLocation);
+#else
+    QString dataDir = QStandardPaths::standardLocations(QStandardPaths::DataLocation).first();
+#endif
+
+    QFileDialog dialog(this, tr("Load sets/cards"));
+    dialog.setDirectory(dataDir);
+    dialog.setNameFilters(TabDeckEditor::fileNameFilters);
+    if (!dialog.exec())
+        return;
+
+    QString fileName = dialog.selectedFiles().at(0);
+
+    if (!QFile::exists(fileName)) {
+        DlgAddSetResult dlg(this, false, QString("Selected file cannot be found."));
+        dlg.exec();
+        return;
+    }
+
+    QDir dir(dataDir + "/customsets");
+    int nextPrefix = getNextCustomSetPrefix(dir);
+
+    bool res = QFile::copy(
+        fileName, dir.absolutePath() + "/" + (nextPrefix > 9 ? "" : "0") +
+        QString::number(nextPrefix) + "." + QFileInfo(fileName).fileName()
+    );
+
+    DlgAddSetResult dlg(this, res, QString());
+    dlg.exec();
+}
+
+int TabDeckEditor::getNextCustomSetPrefix(QDir dataDir) {
+    QStringList files = dataDir.entryList();
+    int maxIndex = 0;
+
+    QStringList::const_iterator filesIterator;
+    for (filesIterator = files.constBegin(); filesIterator != files.constEnd(); ++filesIterator) {
+        int fileIndex = (*filesIterator).split(".").at(0).toInt();
+        if (fileIndex > maxIndex)
```

Comment:
It looks like there's an `std` function for getting max index; http://en.cppreference.com/w/cpp/algorithm/max_element